redis/connection.py

```diff
@@ -1143,15 +1241,20 @@ def _error_message(self, exception):
 FALSE_STRINGS = ("0", "F", "FALSE", "N", "NO")
 
 
-def to_bool(value):
+def to_bool(value: Optional[str]) -> Optional[bool]:
     if value is None or value == "":
         return None
-    if isinstance(value, str) and value.upper() in FALSE_STRINGS:
+    if value.upper() in FALSE_STRINGS:
```

Comment:
Please do not reimplement functionality or make a potentially breaking change when it comes to adding typing support; we don't know what developers are doing, and not everyone likes typing and thus won't follow the rules.

Reply:
Hi, thanks for the feedback. I removed this code snippet as Mypy was complaining about redundant code (`value` will always be string in this code snippet, so there's no reason to check if it's an instance of it).

> we don't know what developers are doing

Thanks for the sincerity, but that part of the sentence really worries me. If you guys don't know what's being put in the code itself, it can be inferred that bad or even malicious code can be inserted into the library (of course, I'm exaggerating a lot, but I think you get the point).

In Python, it is inferred that we are [consenting adults](https://mail.python.org/pipermail/tutor/2003-October/025932.html), this means that the types passed are considered to be the correct ones, and if you are not sure about this, it is safer to create a CI by running a type checker on every commit/pull request (and require every pull request to be properly typed).

If you want, I can create a workflow that runs a type checker in this pull request. That way you guys will be sure that the types are correct.

Reply:
haha I'm sure the types seem fine based on which commands use this util. We know what the function is for. What we don't know is whether a user has imported the function for themselves.

Reply:
> What we don't know is whether a user has imported the function for themselves.

I'm not sure I understand this part of the answer. By "user" do you mean the person who will be using the library? If yes, this answer makes no sense. It's the user's responsibility to pass the correct type to the parameter, and this function isn't even in the documentation so there's no chance of anyone using it (that's why I considered it as an internal).